### PR TITLE
Expose more info with unknown exception

### PR DESCRIPTION
**Pull request by Renkai**
Improves error logs shown in https://github.com/apache/pulsar/issues/9266
How to fix that issue will be decided after more information exposed

### CODE REVIEW

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java [hunk 1]
```diff
@@ -117,6 +115,7 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
             if (!(exception instanceof TooManyRequestsException)) {
                 log.warn("[{}][{}] read failed from ledger at position:{} : {}", cursor.ledger.getName(),
                         cursor.getName(), readPosition, exception.getMessage());
+                log.warn("error caused by: ", exception.getCause());
```
Comment (sijie) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java:118]: Why not just add the stacktrace to the previous `log.warn` statement?

Comment (Renkai) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java:118]: @sijie 
The slf4j API with `Throwable` only support one message string and one throwable parameter, but we still have other info to print. https://github.com/qos-ch/slf4j/blob/master/slf4j-api/src/main/java/org/slf4j/Logger.java#L602

By the way, the logger supports some advanced features like fold stacktrace when printing too frequently, so we'd better not format stacktrace ourselves.

Comment (merlimat) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java:118]: The last arg is always interpreted as an exception if it's of `Throwable` type

Comment (Renkai) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java:118]: Cool, I will move it up.

**Comment by 315157973** [comment]
LGTM

**Review by 315157973 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Review by merlimat — APPROVED** [review]

**Pull request merged**